users/serializers.py

```diff
@@ -252,10 +256,17 @@ def validate(self, data):
             ) as exc:
                 log.exception("Unable to create user account", exc)  # noqa: PLE1205, TRY401
                 raise serializers.ValidationError(USER_REGISTRATION_FAILED_MSG)  # noqa: B904
-
-            if openedx_validation_msg:
-                raise serializers.ValidationError({"username": openedx_validation_msg})
-
+            if openedx_validation_msg_dict["username"]:
+                raise serializers.ValidationError(
+                    {
+                        "username": OPENEDX_ACCOUNT_CREATION_VALIDATION_MSGS_MAP.get(
+                            openedx_validation_msg_dict["username"]
+                        )
+                    }
+                )
+            if openedx_validation_msg_dict["email"]:
+                # there is no email form field at this point, but we are still validating the email address
+                raise serializers.ValidationError(openedx_validation_msg_dict["email"])
```

Comment:
We wanted to display the custom message we defined instead of the message we are getting from Open edX API.

Reply:
This message is never going to show to the user anyway since we want to add this validation to STEP 1. So I don't see a reason to change this. The message from edX API is informative and accurate.

Reply:
I am unsure if we will do that sooner as per this comment https://github.com/mitodl/mitxonline/pull/2417#issuecomment-2435255463. 

Also, we should have kept it consistent. Now we are displaying a custom message for the username but not for the email. Also, the custom message that we defined and added to the OPENEDX_ACCOUNT_CREATION_VALIDATION_MSGS_MAP is unused now.

Reply:
This looks good to me:
<img width="1776" alt="Screenshot 2024-11-06 at 9 02 08 AM" src="https://github.com/user-attachments/assets/0201441f-3555-49d8-9230-46e82fc6cedc">